src/boat/private/BoatConfig.nim

```diff
@@ -0,0 +1,99 @@
+##
+## Config
+## ======
+## extendable interface for creating, parsing and saving boat configs
+
+##[
+## TLDR
+- come back later
+
+todos
+-----
+- cmd to parse arbitrarily, e.g. boat conf load ./some/dir
+- cmd to see current captains: e.g. boat conf list -> X, Y, Z
+]##
+
+from ../../../bdd import tddError
+
+import std/[
+  json,
+  os,
+  strutils,
+  ]
+
+import
+  BoatConstants,
+  BoatErrors,
+  BoatConfigType,
+  FileManager
+
+# consumers can retrieve the parsed Config and path on disk
+# but only internal functions should be able to set it
+export BoatConfigType.BoatConfig, parsed, parsedPath
+
+var captainsLog* {.global.} = %* {} ## \
```

Comment:
pretty sure this should be a ref

Reply:
unsure if we actually need {.global.} if its a ref

.... and pretty sure captainslog should be isolated to a captainslog.nim file and imported